arbnode/maintenance.go

```diff
@@ -112,47 +129,52 @@ func wentPastTimeOfDay(before time.Time, after time.Time, timeOfDay int) bool {
 	return prevMinutes < dbCompactionMinutes && newMinutes >= dbCompactionMinutes
 }
 
-func (c *MaintenanceRunner) maybeRunMaintenance(ctx context.Context) time.Duration {
-	config := c.config()
+func (mr *MaintenanceRunner) maybeRunMaintenance(ctx context.Context) time.Duration {
+	config := mr.config()
 	if !config.enabled {
 		return time.Minute
 	}
+
 	now := time.Now().UTC()
-	if wentPastTimeOfDay(c.lastCheck, now, config.minutesAfterMidnight) {
-		log.Info("attempting to release sequencer lockout to run database compaction", "targetTime", config.TimeOfDay)
-		if c.seqCoordinator == nil {
-			c.runMaintenance()
-		} else {
-			// We want to switch sequencers before running maintenance
-			success := c.seqCoordinator.AvoidLockout(ctx)
-			defer c.seqCoordinator.SeekLockout(ctx) // needs called even if c.Zombify returns false
-			if success {
-				// We've unset the wants lockout key, now wait for the handoff
-				success = c.seqCoordinator.TryToHandoffChosenOne(ctx)
-				if success {
-					c.runMaintenance()
-				}
-			}
-		}
+	mr.lastCheck = now
```

Comment:
Updating this before the wentPastTimeOfDay check means that the check will never pass

Reply:
Renamed variable to express meaning better and setting only when maintenance runs.